### ISSUE TITLE
PCC Drop Observed in BatchNorm2d When Running Conv2d + BatchNorm2d Together (PCC = 0.6252649726816831)

### ISSUE THREAD

**Issue by kamalrajkannan78** [tenstorrent/tt-metal#38634]
### Describe the bug 

- A PCC drop is observed in BatchNorm2d when Conv2d and BatchNorm2d are executed together, resulting in a PCC of `0.6252649726816831`.
- The experiments below confirm this observation.
- Centernet hourglass variant(red model) is affected by this error - https://github.com/tenstorrent/tt-xla/issues/1505#issuecomment-3965365444

### Summary of Experiments

- **EXP 1** - ttnn.conv2d + ttnn.batch_norm -> `0.6252649726816831`
- **EXP 2** – Intermediate Output Validation - As an additional check, intermediate (Conv2d) and final outputs were compared instead of only validating the final output
  - Conv Output PCC: `0.9996460723330881`
  - Final Output PCC: `0.6252649726816831`
- **EXP 3** - ttnn.conv2d -> `0.9996460723330881`
- **EXP 4** - ttnn.batch_norm -> `0.9999920791277928` 
- It appears that the small PCC drop introduced by Conv2d is significantly amplified by BatchNorm2d, leading to a much larger drop in the final output. This might be similar to the previously reported [case](https://github.com/tenstorrent/tt-metal/issues/36394#issuecomment-3810036978). 
- @bbradelTT could you please take a look and confirm whether my observation is correct?

### Repro

```
git checkout kkannan/feb26_pcc_drop_conv2d_bn2d_metal
git submodule update --init --recursive

git lfs pull --include '*.pt *.tensorbin'
# As an alternate option you can unzip the input_tensors.zip for required input tensors

# exp runs
pytest tests/ttnn/unit_tests/operations/conv2d_bn2d/test_exp1.py -svv
pytest tests/ttnn/unit_tests/operations/conv2d_bn2d/test_exp2.py -svv
pytest tests/ttnn/unit_tests/operations/conv2d_bn2d/test_exp3.py -svv
pytest tests/ttnn/unit_tests/operations/conv2d_bn2d/test_exp4.py -svv
```

### Logs

- [input_tensors.zip](https://github.com/user-attachments/files/25575039/input_tensors.zip)
- [feb26_exp1.log](https://github.com/user-attachments/files/25575047/feb26_exp1.log)
- [feb26_exp2.log](https://github.com/user-attachments/files/25575049/feb26_exp2.log)
- [feb26_exp3.log](https://github.com/user-attachments/files/25575050/feb26_exp3.log)
- [feb26_exp4.log](https://github.com/user-attachments/files/25575048/feb26_exp4.log)

**Comment by kamalrajkannan78** [comment]
> Talked to [@kamalrajkannan78](https://github.com/kamalrajkannan78) and bfloat16 is used. Based on previous issues, that leads to bad pcc because the values are too close to the mean and subtracting from the mean leads to values getting rounded down to 0 in various patterns that result in bad pcc. You need to use float32 to get good pcc.

- I tried with fp32 inputs, but the PCC actually got worse:
- EXP 1 (ttnn.conv2d + ttnn.batch_norm): `0.5306809735433204` with fp32 vs `0.6252649726816831` with bfloat16
- This suggests the root cause here is different from the [previous ticket](https://github.com/tenstorrent/tt-metal/issues/36394#issuecomment-3810036978), where fp32 inputs resolved the drop. In this case, fp32 didn't help - it made things worse.
- @bbradelTT could you please take a closer look at the bfloat16 case mentioned in description? The isolation experiments (EXP 1–4) in the description should help narrow it down.

### Repro

```
git checkout kkannan/feb26_pcc_drop_conv2d_bn2d_metal
git submodule update --init --recursive

git lfs pull --include '*.tensorbin'
# As an alternate option you can unzip the fp32_tensors.zip for required input tensors

# exp runs
pytest tests/ttnn/unit_tests/operations/conv2d_bn2d_fp32/test_exp1_fp32.py
```

### Logs

- [fp32_tensors.zip](https://github.com/user-attachments/files/25598213/fp32_tensors.zip)
- [feb27_exp1_fp32.log](https://github.com/user-attachments/files/25598183/feb27_exp1_fp32_f.log)

**Comment by pavlejosipovic** [comment]
that is affecting only activation function if it is present
